### ISSUE TITLE
Flexible handling of (multiple) entry points in Palladium's config

### ISSUE THREAD

**Issue by alattner** [ottogroup/palladium#26]
Right now there exist /alive and only one /predict entry points and there is no nice way to specify an individual entry point or even multiple entry points. We've had some cases where we wanted to have more than one /predict variant in one Palladium instance and have implemented it by cloning the predict function and asigning a new entry point.

The solution should allow for specifying multiple predict service instances (and entry point names).

**Comment by dnouri** [comment]
How do these variants look like?  Are these really clones of `/predict` just with different configuration?

**Comment by alattner** [comment]
In most cases it would be using the same model in two different ways, e.g., calls with different parameters where some preprocessing or lookup is performed. Maybe we could use a factory to create functions with specific entry points and predict services and use it in Palladium's config like this:
```
    'predict_service_1': {
        '__factory__': 'palladium.server.create_predict_service',
        'entry_point_name': 'predict',
        'predict_service': {
            '__factory__': 'myproject.server.PredictService',
            'mapping': [
                ('text', 'str'),
                ],
            }
        },

    'predict_service_2': {
        '__factory__': 'palladium.server.create_predict_service',
        'entry_point_name': 'predict-id',
        'predict_service': {
            '__factory__': 'myproject.server.PredictServiceId',
            'mapping': [
                ('id', 'str'),
                ],
            }
        },
```

**Comment by alattner** [comment]
Right now I am working on a solution where entry points can be defined in the config like this (instead of the approach mentioned above):
```
    'entry_points': {
        '__factory__': 'palladium.server.register_predict_services',
        'mapping': {
            '/predict1': {
                'predict_service_name': 'predict_service',
                'decorator_list_name': 'predict_decorators',
            },
            '/predict2': {
                'predict_service_name': 'predict_service',
                'decorator_list_name': 'predict_decorators',
            },
        }
    },
```
The definition of PredictServices and decorator_lists remains unchanged, but if one wants to provide more than one entry point, it can be done using such an `entry_points` section.

**Comment by alattner** [comment]
Merged new feature with pull request #32. It can be used in Palladium's config like that:
```
    'predict_service1': {
        '__factory__': 'mypackage.server.PredictService',
        'mapping': [
            ('sepal length', 'float'),
            ('sepal width', 'float'),
            ('petal length', 'float'),
            ('petal width', 'float'),
            ],
	'entry_point': '/predict',
	'decorator_list_name': 'predict_decorators',
        }
    'predict_service2': {
        '__factory__': 'mypackage.server.PredictServiceID',
        'mapping': [
            ('id', 'int'),
            ],
	'entry_point': '/predict-by-id',
	'decorator_list_name': 'predict_decorators_id',
        }
```
